components/email_aliases/email_aliases_service_unittest.cc

```diff
@@ -345,3 +342,394 @@ TEST_F(EmailAliasesServiceTimingTest, VerifyResult_StopsAfterMaxDuration) {
+    }
+    url_loader_factory_.AddResponse(manage_url.Resolve("?status=active").spec(),
+                                    refresh_body.value_or("[]"));
+
+    base::MockCallback<
+        base::OnceCallback<void(base::expected<std::monostate, std::string>)>>
+        cb;
+    bool called = false;
+    base::expected<std::monostate, std::string> result_out;
+    EXPECT_CALL(cb, Run(testing::_))
+        .Times(1)
+        .WillOnce([&](base::expected<std::monostate, std::string> res) {
+          called = true;
+          result_out = std::move(res);
+        });
+    service_->UpdateAlias(alias_email, /*note=*/std::string("note"), cb.Get());
+    EXPECT_TRUE(base::test::RunUntil([&]() { return called; }));
+    if (wait_for_update) {
+      EXPECT_TRUE(observer_.WaitForAliasUpdateCount(1));
+    }
+    return result_out;
+  }
+
+  base::expected<std::monostate, std::string> CallDeleteAliasWith(
+      const std::string& alias_email,
+      const std::optional<std::string>& delete_body,
+      const std::optional<int>& net_error = std::nullopt) {
+    const GURL manage_url =
+        EmailAliasesService::GetEmailAliasesServiceBaseURL();
+    if (net_error.has_value()) {
+      network::URLLoaderCompletionStatus completion(*net_error);
+      auto head = network::mojom::URLResponseHead::New();
+      url_loader_factory_.AddResponse(manage_url, std::move(head),
+                                      /*content=*/"", completion);
+    } else {
+      url_loader_factory_.AddResponse(manage_url.spec(),
+                                      delete_body.value_or(""));
+    }
+    url_loader_factory_.AddResponse(manage_url.Resolve("?status=active").spec(),
+                                    "[]");
+
+    base::MockCallback<
+        base::OnceCallback<void(base::expected<std::monostate, std::string>)>>
+        cb;
+    bool called = false;
+    base::expected<std::monostate, std::string> result_out;
+    EXPECT_CALL(cb, Run(testing::_))
+        .Times(1)
+        .WillOnce([&](base::expected<std::monostate, std::string> res) {
+          called = true;
+          result_out = std::move(res);
+        });
+    service_->DeleteAlias(alias_email, cb.Get());
+    EXPECT_TRUE(base::test::RunUntil([&]() { return called; }));
+    EXPECT_TRUE(observer_.WaitForAliasUpdateCount(1));
+    return result_out;
+  }
+};
+
+// ================= Parameterized cases (shared) =================
+
+namespace {
+
+// Expected alias value used by Generate tests (success path).
+constexpr char kExpectedGeneratedAlias[] = "mock-1234@bravealias.com";
+
+// Build the success response body for Generate tests using the expected alias.
+std::string MakeGenerateAliasSuccessBody() {
+  return std::string("{\"message\":\"created\",\"alias\":\"") +
+         kExpectedGeneratedAlias + "\"}";
+}
+
+struct EmailAliasesApiCase {
+  const char* name;
+  std::optional<std::string> body;
+  std::optional<int> net_error;
```

Comment:
As per my other comment above, you wouldn't need `net_error`.

Reply:
See simplification commit from @szilardszaloki here: https://github.com/brave/brave-core/commit/ff23527c9376c84387d8ec7d5166d53715934a9f